crates/dbsp/src/circuit/runtime.rs

```diff
@@ -638,28 +560,51 @@ impl Runtime {
             .write()
             .map(|mut guard| *guard = Some(panic_info));
     }
+
+    /// Spawn a new thread using `builder` and `f`. If the current thread is
+    /// associated with a runtime, then the new thread will also be associated
+    /// with the same runtime and worker index.
+    pub(crate) fn spawn_background_thread<F>(builder: Builder, f: F) -> Thread
+    where
+        F: FnOnce() + Send + 'static,
+    {
+        let runtime = Self::runtime();
+        let worker_index = Self::worker_index();
+        let join_handle = builder
+            .spawn(move || {
+                RUNTIME.with(|rt| *rt.borrow_mut() = runtime);
+                WORKER_INDEX.set(worker_index);
+                f()
+            })
+            .unwrap_or_else(|error| {
```

Comment:
isn't this pattern the same as .expect()

Reply:
It would be with a plain string literal argument but I'd have to use `expect(&format!(...))` here and it would always format and allocate even in the common case of not-an-error.